ee/backend/app/src/main/kotlin/io/tolgee/ee/service/TaskAssigneeUpdatedNotificationListener.kt

```diff
@@ -0,0 +1,53 @@
+package io.tolgee.ee.service
+
+import io.tolgee.events.OnEntityPreUpdate
+import io.tolgee.model.Notification
+import io.tolgee.model.UserAccount
+import io.tolgee.model.task.Task
+import io.tolgee.repository.NotificationRepository
+import io.tolgee.util.Logging
+import io.tolgee.util.logger
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+class TaskAssigneeUpdatedNotificationListener(
+  private val notificationRepository: NotificationRepository,
+) : Logging {
+  @EventListener
+  fun onTaskAssigneeChange(event: OnEntityPreUpdate) {
+    val entity = event.entity
+
+    if (entity !is Task)
+      return
+
+    getNewAssignees(event, entity).forEach {
+      notificationRepository.save(
+        Notification(
+          user = it,
+          linkedTask = entity,
+          originatingUser = entity.author,
+          project = entity.project,
+        )
+      )
+    }
+  }
+
+  private fun getNewAssignees(event: OnEntityPreUpdate, entity: Task) : List<UserAccount> {
+    val assigneesIndex = event.propertyNames?.indexOf("assignees")
```

Comment:
It might maybe make more sense to trigger the event manually in the method which takes care of the creating / updating task to get properly typed event data. 

Also there is the OnProjectActivityEvent, which I would like stick to using. It's executed before the commit, so it should be OK for this case. However, I would still probably prefer to trigger it manually for this case if you don't really have super good reasons for this.